### PR TITLE
Improve wayfinding for various Volto audiences

### LINKED ISSUE #5730: Delete Developer roadmap from docs?
@tiberiuichim you worked on the original document. Should we delete [Developer roadmap](https://6.docs.plone.org/volto/getting-started/roadmap.html)? I think it is redundant to everything under [Contributing to Volto](https://6.docs.plone.org/volto/contributing/index.html), https://plone.org/why-plone/roadmap, and https://github.com/orgs/plone/projects/33/views/1.

### PR DESCRIPTION
**Pull request by stevepiercy**
Closes #5730

See also https://github.com/plone/documentation/pull/1630

### CODE REVIEW

**Comment by stevepiercy** [comment]
https://github.com/plone/volto/actions/runs/8082176684/job/22082399377?pr=5809#step:4:224 will be fixed after this PR is merged and deployed to the main documentation. It can be safely ignored.

**Comment by ichim-david** [comment]
@stevepiercy I think I like better how fast the "other learning resources" were presented in the current docs as opposed to how it is now towards the end of the table of contents.
So I would add it after the Development -> Overview page.
I also don't know if it's good enough to rename the Recipes to Development and have this section first.
If it is then it should probably have some info structured better.

I don't have a good suggestion here as I would have to look in more depth on how the tutorials were instructed from the trainings, then what we have and then to give more targeted suggestions. 
Hopefully the others have better suggestions but this was my initial jab at saying something until I can have more suggestive comments.

**Comment by stevepiercy** [comment]
Check the [linked issue](https://github.com/plone/volto/issues/5730) to get @tiberiuichim's intent for "Developer Roadmap", renamed to "Overview". I tried to preserve this intent.

This PR is also intended as another step toward [Diátaxis](https://diataxis.fr/), hence the creation of a new section for Learning/Tutorials. Because learning does not target a specific developers/integrators audience, it does not belong under Development. In addition, Development is more of a How-To Guides collection.

From our Matomo website statistics, I got two interesting reports (see below). From that data, people who want to perform a specific development task in Volto is the largest segment. Thus the reordering of the navigation items is to better serve that purpose. Note that GSoC is one of the largest referrers to Volto docs, which skews the "contributing" section popularity.

The most popular entry sections in Volto docs are, in order, the following.

-   contributing
-   /index.html
-   recipes
-   blocks
-   configuration
-   addons
-   user-manual
-   getting-started
-   client
-   deploying
-   release-notes
-   upgrade-guide
-   theming

The most popular sections of Volto docs that are visited after performing a search are, in order, the following.

-   contributing
-   recipes
-   addons
-   configuration
-   client
-   blocks
-   deploying
-   release-notes
-   upgrade-guide
-   theming
-   user-manual
-   getting-started

File: docs/source/index.md
Comment (davisagli) [inline docs/source/index.md:32]: how about
```suggestion
An integrator is someone who uses Volto to build a website.
```

It can be for yourself, rather than for a client...

**Review by davisagli — APPROVED** [review]
@stevepiercy This reorganization looks like a helpful step to me!

File: docs/source/index.md
Comment (stevepiercy) [reply docs/source/index.md:32]: As well as for an employer. However, which term would better describe Plone?

- website
- web application

I lean toward the latter, as it implies more than just static content, but includes workflows and moving data from point A to B.

Comment (davisagli) [reply docs/source/index.md:32]: @stevepiercy Not sure we want to get into that here. Just say "to build a project"?

Comment (stevepiercy) [reply docs/source/index.md:32]: Yup, that works.

Comment (stevepiercy) [reply docs/source/index.md]: Fixed in [ce5da24](https://github.com/plone/volto/pull/5809/commits/ce5da243151845ffd060e11e42798c18c734f456)

**Comment by stevepiercy** [comment]
I'd really like @tiberiuichim's review on this pull request, as he was the creator of the original roadmap.

I also would like @sneridagh's review, as a primary maintainer.

For convenience, preview links for the significant changes (most changes are fixing cross-references and moving files):

- [Frontend](https://deploy-preview-5809--volto.netlify.app/)
- [Development](https://deploy-preview-5809--volto.netlify.app/development/index.html) index page
- [Overview](https://deploy-preview-5809--volto.netlify.app/development/overview.html)
- [Learning resources](https://deploy-preview-5809--volto.netlify.app/tutorials/index.html) new section

**Review by sneridagh — APPROVED** [review]

**Comment by stevepiercy** [comment]
Thank you, everyone!

**Pull request merged**
